crates/trie/sparse/src/state.rs

```diff
@@ -521,11 +530,17 @@ impl<F: BlindedProviderFactory> SparseStateTrie<F> {
     ///
     /// If the trie has not been revealed, this function reveals the root node and returns its hash.
     pub fn root(&mut self) -> SparseStateTrieResult<B256> {
+        // record revealed node metrics
+        self.metrics.record();
```

Comment:
should be fine to run only at the end of computation, and not on every `update_rlp_node_level`